src/app/_services/assets.service.ts

```diff
@@ -701,7 +701,7 @@ export class AssetService {
                 }
             });
 
-            if (params['startDate'] && params['endDate']){
+            if (params['startDate'] || params['endDate']){
```

Comment:
IMO the code below becomes a little obscure if we're not expecting integer values. In particular, we do math operations on undefined values:

```
                    earliest : {
                        date : Math.abs(params['startDate']),
                        era : params['startDate'] < 0 ? 'BCE' : 'CE'
                    },
```

Is there a way to clarify how we're handling the `undefined` case?

Reply:
With javascript, undefined becomes 0... Which is weird. This ticket was to have advanced search act the same way the side-facet bar on the browse page works. which defaults to year 0 for any year not entered. We have a ticket SRCHX-406 which will change this behavior

Reply:
Oh, yeah, I don't doubt that it works. I just thought it looked obscure because it implicitly coerces `undefined`.

Reply:
For example, the following looks equivalent, but gives a different result because of the hidden case:

```
era : params['startDate'] >= 0 ? 'CE' : 'BCE'
```